osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/dlmsobjectconfig/model/DlmsObject.java

```diff
@@ -31,10 +32,14 @@ public int getClassId() {
         return this.classId;
     }
 
-    public String getObisCode() {
+    public String getObisCodeAsString() {
         return this.obisCode;
     }
 
+    public ObisCode getObisCode() {
+        return new ObisCode(this.obisCode);
+    }
+
```

Comment:
Hmm, ik vind dit zo op het eerste gezicht geen verbetering. Er is maar 1 aanroep naar de methode getObisCode nu:
```
        if (dlmsObject.isPresent()) {
            return dlmsObject.get().getObisCode();
        } else {
            throw new ProtocolAdapterException("Dlms object not found in config, type: " + type + ", medium: " + filterMedium);
        }
```

Kan daar niet gewoon een instance van ObisCode worden gemaakt met de obis code string als input?

Reply:
I would like to move the responsibility of converting ObisCodes to Strings in this class. In the (D)SMR many gas objects are identified with an obiscode like "0.x.24.3.0.255", where x stands for mbus channel. An "x" is not allowed by the jDLMS ObisCode class, which lead to using a String for the obis code in the DlmsObject class and all other classes doing a conversion with channels. This conversion could be done in this class. I didn't do it yet, because that would've been too much for this story.

Reply:
Fair enough. I do think such a conversion should reside in a separate class when the time comes.

+1 for switching us to English